packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -172,7 +172,7 @@ function UnconnectedNavigatorProvider(
 					skipFocus,
 				};
 
-				if ( prevLocationHistory.length < 1 ) {
+				if ( prevLocationHistory.length < 1 || isBack ) {
```

Comment:
This change makes it so that calling `navigator.goToParent()` or `navigator.goTo( ...,  { isBack: true } )` will always clear the history. It's needed because otherwise if you call `goBack()` after a call to `goToParent()` you will be back at the child page.

An alternative to this could be adding an explicit `resetHistory` param to `goTo` and `goToParent`.

Reply:
> This change makes it so that calling navigator.goToParent() or navigator.goTo( ...,  { isBack: true } ) will always clear the history. It's needed because otherwise if you call goBack() after a call to goToParent() you will return to the child page.

Actually if I think about the Navigator component on its own, that expectation is correct. If I click goBack after goToParent, I expect to go to the child personally.

Reply:
Fair enough. I've tweaked this so that you pass `{ replace: true }` to `goTo` or `goToParent` if you don't want to add to history.